### PR TITLE
[GAL-4258] Improve token global sync

### PR DESCRIPTION
**Pull request by jakzaizzat**
### Summary of Changes

Move the token sync state to global context. So that, we can reduce redundant request on sync

### Demo or Before/After Pics

https://github.com/gallery-so/gallery/assets/4480258/3dc87c7c-ad45-4e59-8eea-e2e6c4ffb2f2

### Testing Steps

If you the token already syncing, the refresh button should be un-clickable

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/contexts/SyncTokensContext.tsx [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+import { useToastActions } from './ToastContext';
+
+type SyncTokensActions = {
+  syncTokens: (chain: Chain) => void;
+  isSyncing: boolean;
+};
+
+const SyncTokensActionsContext = createContext<SyncTokensActions | undefined>(undefined);
+
+export const useSyncTokenstActions = (): SyncTokensActions => {
```
Comment (kaitoo1) [inline apps/mobile/src/contexts/SyncTokensContext.tsx:16]: small typo `Tokenst`

File: apps/mobile/src/contexts/SyncTokensContext.tsx [hunk 2]
```diff
@@ -66,7 +86,15 @@ export default function useSyncTokens() {
...
   );
 
-  return useMemo(() => {
+  const value = useMemo(() => {
     return { syncTokens: sync, isSyncing };
   }, [isSyncing, sync]);
-}
+
+  return (
+    <SyncTokensActionsContext.Provider value={value}>{children}</SyncTokensActionsContext.Provider>
```
Comment (kaitoo1) [inline apps/mobile/src/contexts/SyncTokensContext.tsx:94]: can self close this?

Comment (kaitoo1) [reply apps/mobile/src/contexts/SyncTokensContext.tsx:94]: or include children?

**Review by kaitoo1 — APPROVED** [review]
niiiice clutch refactor!! 🔥

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**